src/axom/multimat/multimat.hpp

```diff
@@ -0,0 +1,765 @@
+    const std::string& field_name);
+
+  /**
+   * \brief Get the volume fraction field
+   */
+  Field2D<double>& getVolfracField();
+
+  /**
+   * \brief Get a set of index for a Subfield.
+   * \detail for a cell-dominant layout, this is equivalent to getting the set
+   * of material presented in a cell. Vise versa, for a material-dominant
+   * layout, this returns a set of cell that contains a material.\n
+   * getMatInCell() and getCellContainingMat() are layout specific
+   * calls for this function.
+   *
+   * \param idx the index of the subfield
+   * \return IdSet A set of index in the subfield
+   */
+  IdSet getSubfieldIndex(int idx);
+
+  /** Cell-dominant version of getSubfieldIndex() **/
+  IdSet getMatInCell(int cell_id);
+  /** Material-dominant version of getSubfieldIndex() **/
+  IdSet getCellContainingMat(int mat_id);
+
+  /**
+   * \brief Get a set to index a Subfield in a Field2D.
+   * \detail This returns a set of indices that can be used to directly access
+   * a subfield in a Field2D without having to retrieve a Subfield first.\n
+   * Accessing the Field2D using this indexing set should be done with the
+   * bracket operator.
+   *
+   * e.g.
+   * \code
+   *   IndexSet indexing_set = getSubfieldIndexingSet( idx );
+   *   for( int i : indexing_set );
+   *       value = field2d[ i ];
+   * \endcode
+   *
+   * Or to account for multiple components:
+   * \code
+   *   IndexSet indexing_set = getSubfieldIndexingSet( idx );
+   *   for( int i : indexing_set );
+   *       for( int c = 0; c < numComp; c++)
+   *           value = field2d[ i * numComp + c ];
+   * \endcode
+   *
+   * getIndexingSetOfCell() and getIndexingSetOfMat() are layout specific
+   * calls for this function.
+   *
+   * \param idx The index of the subfield
+   * \return IndexSet A Set of index for the subfield
+   */
+  IndexSet getSubfieldIndexingSet(int idx,
+                                  DataLayout layout,
+                                  SparsityLayout sparsity);
+
+  /** Cell-dominant version of getSubfieldIndexingSet() **/
+  IndexSet getIndexingSetOfCell(int cell_id, SparsityLayout sparsity);
+  /** Material-dominant version of getSubfieldIndexingSet() **/
+  IndexSet getIndexingSetOfMat(int mat_id, SparsityLayout sparsity);
+
+  /** Return the number of material this object holds **/
+  int getNumberOfMaterials() const { return m_nmats; };
+  /** Return the number of cells this object holds **/
+  int getNumberOfCells() const { return m_ncells; }
+
+  //Layout modification functions
+
+  void convertFieldLayout(int field_idx, SparsityLayout, DataLayout);
+
+  void convertFieldToSparse(int field_idx);
+  void convertFieldToDense(int field_idx);
+  SparsityLayout getFieldSparsityLayout(int field_idx);
+
+  void transposeField(int field_idx);
+  void convertFieldToMatDom(int field_idx);
+  void convertFieldToCellDom(int field_idx);
+  DataLayout getFieldDataLayout(int field_idx);
```

Comment:
All these methods operate on a `field_idx`, but the `get*2DField()` methods take in a string for the field name instead: any reason for the difference?